src/QMCWaveFunctions/TrialWaveFunction.cpp

```diff
@@ -564,9 +564,9 @@ void TrialWaveFunction::mw_evalGrad(const RefVectorWithLeader<TrialWaveFunction>
   const int num_wfc             = wf_leader.Z.size();
   auto& wavefunction_components = wf_leader.Z;
 
-  TWFGrads<CT> grads_z(num_wf);
   for (int i = 0; i < num_wfc; i++)
   {
+    TWFGrads<CT> grads_z(num_wf);
     ScopedTimer localtimer(wf_leader.WFC_timers_[VGL_TIMER + TIMER_SKIP * i]);
     const auto wfc_list(extractWFCRefList(wf_list, i));
     wavefunction_components[i]->mw_evalGrad(wfc_list, p_list, iat, grads_z);
```

Comment:
Need a bit change. mw_evalGrad is not expected to depends on previous value of grads_z. This is different from single walker API. At
https://github.com/QMCPACK/qmcpack/blob/27ff296dda38c127f120f6356c268bd5bb268e7a/src/QMCWaveFunctions/WaveFunctionComponent.cpp#L98
Set `spingrad_now[iw]` to zero before calling `evalGradWithSpin`. `grad_now[iw]` has been directly overwritten.

Reply:
reverted TrialWaveFunction back and changed in WaveFunctionComponent.cpp